src/values/AbstractObjectValue.js

```diff
@@ -312,16 +318,18 @@ export default class AbstractObjectValue extends AbstractValue {
       invariant(ob2 instanceof ObjectValue || ob2 instanceof AbstractObjectValue);
       let d1 = ob1.$GetOwnProperty(P);
       let d2 = ob2.$GetOwnProperty(P);
-      if (d1 === undefined || d2 === undefined || !equalDescriptors(d1, d2)) {
+      if (d1 !== undefined && d2 !== undefined && !equalDescriptors(d1, d2)) {
```

Comment:
This models the case when one descriptor is empty and not the other. In a follow up I'd like to model this as joinedDescriptors to support two non-equal descriptors too. There's an unrelated bug in properties.js regarding that.

This becomes important after this PR because $Get on each of the conditional objects now calls $GetOwnProperty on the Receiver which is both objects, instead of on the unwrapped object. So the $Get now gets two possible results, instead of one, which the simplifier then cleans up.